pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -430,6 +430,29 @@ CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.label_unnest(label_array anyarray)
 AS $function$ SELECT unnest(label_array) $function$;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.label_unnest(anyarray) to prom_reader;
 
+-- approximate_row_count is used to count the approximate number of rows in a table (non-hypertable).
+CREATE OR REPLACE FUNCTION SCHEMA_PROM.approximate_row_count(table_name REGCLASS)
+    RETURNS BIGINT
+    LANGUAGE SQL
+AS
+$$
+    SELECT reltuples::BIGINT FROM pg_class WHERE oid=table_name;
+$$ STABLE PARALLEL SAFE;
+
+-- safe_approximate_row_count returns the approximate row count of a hypertable if timescaledb is installed
+-- else returns the approximate row count in the normal table. This prevents errors in approximate count calculation
+-- if timescaledb is not installed, which is the case in plain postgres support.
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.safe_approximate_row_count(table_name REGCLASS) RETURNS BIGINT
```

Comment:
This only works on hypertables so the name should reflect that we only use it on data tables.
```suggestion
CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.data_table_approximate_row_count(table_name REGCLASS) RETURNS BIGINT
```

Reply:
> This only works on hypertables

Only the `public.hypertable_approximate_row_count` works on hypertables, not `schema_prom.approximate_row_count`, and hence gave it a generic name.